x/supply/internal/types/permissions.go

```diff
@@ -1,20 +1,60 @@
 package types
 
-import "fmt"
+import (
+	"fmt"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
 
 // permissions
 const (
-	Basic  = "basic"
-	Minter = "minter"
-	Burner = "burner"
+	Basic   = "basic"
+	Minter  = "minter"
+	Burner  = "burner"
+	Staking = "staking"
 )
 
-// validate the input permissions
-func validatePermissions(permission string) error {
-	switch permission {
-	case Basic, Minter, Burner:
-		return nil
-	default:
-		return fmt.Errorf("invalid module permission %s", permission)
+// PermAddr defines the permissions for an address
+type PermAddr struct {
```

Comment:
Maybe we should rename this to `PermissionsForAddress`

Reply:
updating this, going to leave variables as `permAddr` if that is fine?

Reply:
yeah, I don't see a problem with that